enclave/src/main/java/org/hyperledger/besu/enclave/Enclave.java

```diff
@@ -26,101 +28,117 @@
 
 import java.io.IOException;
 import java.net.URI;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 import com.fasterxml.jackson.core.JsonParseException;
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonMappingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import okhttp3.MediaType;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import io.netty.util.AsciiString;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import io.vertx.core.http.HttpMethod;
 
 public class Enclave {
-  private static final Logger LOG = LogManager.getLogger();
+
   private static final ObjectMapper objectMapper = new ObjectMapper();
-  private static final MediaType JSON = MediaType.parse("application/json");
-  private static final MediaType ORION = MediaType.get("application/vnd.orion.v1+json");
+  private static final AsciiString JSON = APPLICATION_JSON;
+  private static final AsciiString ORION = AsciiString.of("application/vnd.orion.v1+json");
 
-  private final URI enclaveUri;
-  private final OkHttpClient client;
+  private final Vertx vertx;
+  private final HttpClient client;
 
   public Enclave(final URI enclaveUri) {
-    this.enclaveUri = enclaveUri;
-    this.client = new OkHttpClient();
+    final HttpClientOptions clientOptions = new HttpClientOptions();
+    clientOptions.setDefaultHost(enclaveUri.getHost());
+    clientOptions.setDefaultPort(enclaveUri.getPort());
+
+    vertx = Vertx.vertx();
+    this.client = vertx.createHttpClient(clientOptions);
   }
 
-  public boolean upCheck() throws IOException {
-    final String url = enclaveUri.resolve("/upcheck").toString();
-    final Request request = new Request.Builder().url(url).get().build();
+  public boolean upCheck() throws ExecutionException, InterruptedException {
```

Comment:
Would of excepted this to throw an EnclaveException?

Reply:
I don't like this function - its not used by production code - only test code. However, I appreciate that its there for a complete interface.
Given it returns a boolean, I think this shouldn't throw at all - should return true if Orion responded, false otherwise...

Reply:
If it's only used by test code I don't think it should even be part of the Enclave